cmd/buildctl/build.go

```diff
@@ -233,11 +234,14 @@ func build(clicontext *cli.Context) error {
 
 	eg.Go(func() error {
 		var c console.Console
-		if !clicontext.Bool("no-progress") {
-			if cf, err := console.ConsoleFromFile(os.Stderr); err == nil {
-				c = cf
-			}
+		progressOpt := clicontext.String("progress")
+		if progressOpt != "auto" && progressOpt != "tty" && progressOpt != "plain" {
+			return errors.New("Invalid progress value")
 		}
+		if cf, err := console.ConsoleFromFile(os.Stderr); err == nil && (progressOpt == "auto" || progressOpt == "tty") {
```

Comment:
In what situations do errors occur here? Is an error only generated if "no console" is present (so, ok to ignore if `"auto"` is selected), or are there other situations?

What if `"tty"` is selected, and an error occurs? Should that error be returned?

Reply:
Also wondering if a `switch` would be slightly clearer; doing so would also prevent running `console.ConsoleFromFile()` if not needed (i.e., if `"plain"` was used as option);

```go
switch clicontext.String("progress") {
case "auto", "tty":
	cf, err := console.ConsoleFromFile(os.Stderr)
	if err != nil {
		return err
	}
	c = cf
case "plain":
default:
	return errors.New("Invalid progress value")
}
```

Reply:
right now, error is not returned if `tty` is option and is failback on `plain`.
I will update it as per suggestion.